02-components/01-CounterButtonComponent/CounterButton.js

```diff
@@ -1,7 +1,23 @@
 export const CounterButton = {
   // Шаблон потребуется отредактировать
-  template: '<button type="button"></button>',
-
+  name: 'CounterButton',
+  template: '<button type="button" @click="counter(count)">{{ count }}</button>',
+  props: {
+   count: {
+     type: Number,
+     default: 0,
+   }
+  },
+  model: {
+    event: 'increment',
+    prop: 'count',
+  },
+  methods: {
+    counter(count) {
+      count++;
```

Comment:
Какую роль играет аргумент функции counter?
Текущее значение можно получить из входного параметра: this.count (менять его, однако, нельзя, за него отвечает родитель).

Reply:
Да, параметр не нужен.
Но можно ли тогда написать так? 
counter() {
      this.count++;
      this.$emit('increment', this.count);
    },

Reply:
Видимо - нет, Vue выдает мне ошибку [Vue warn]: Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed p
roperty based on the prop's value. Prop being mutated: "count"

А параметр я пробрасывала текущее значение

Reply:
Так написать нельзя, так как count - это входной параметр компонента, его в компонент передаёт родитель либо явно (`:count="value"`), либо через `v-model="value"`.

После присваивания нового значения в this.count, он становится более не связанный с родителем, так как получен не от родителя.

И нам не нужно его менять. Нам требуется лишь с событием отдать родителю новое значение, а уже он его вернёт снова через параметр.

От родителя к ребёнку: пропс.
От ребёнка к родителю: событие.

Reply:
А как тогда передать? Завести 
новую переменную?

Reply:
Чтобы передать значение, не обязательно заводить новую переменную, можно передать значение выражения, например, передать `this.count + 1`

Reply:
Я сначала так поняла, что любые операции
 с this.count нельзя производить, 
Теперь вижу разницу между 
this.count+ 1 и this.count++